src/data/quot.lean

```diff
@@ -551,4 +561,60 @@ protected lemma mk'_eq_mk (x : α) : quotient.mk' x = ⟦x⟧ := rfl
 
 end
 
+/-! ### Decidability -/
+
+instance (r : α → α → Prop) (f : α → Prop) (h : ∀ a b, r a b → f a = f b)
+  [hf : decidable_pred f] :
+  decidable_pred (quot.lift f h) :=
+λ q, quot.rec_on_subsingleton q hf
+
+instance (r : α → α → Prop) (s : β → β → Prop) (f : α → β → Prop)
+  (ha : ∀ a b₁ b₂, s b₁ b₂ → f a b₁ = f a b₂) (hb : ∀ a₁ a₂ b, r a₁ a₂ → f a₁ b = f a₂ b)
+  [hf : Π a, decidable_pred (f a)] (q₁ : quot r) :
+  decidable_pred (quot.lift₂ f ha hb q₁) :=
+λ q₂, quot.rec_on_subsingleton₂ q₁ q₂ hf
+
+instance (r : α → α → Prop) (q : quot r) (f : α → Prop) (h : ∀ a b, r a b → f a = f b)
+  [decidable_pred f] :
+  decidable (quot.lift_on q f h) :=
+quot.lift.decidable_pred _ _ _ _
+
+instance (r : α → α → Prop) (s : β → β → Prop) (q₁ : quot r) (q₂ : quot s) (f : α → β → Prop)
+  (ha : ∀ a b₁ b₂, s b₁ b₂ → f a b₁ = f a b₂) (hb : ∀ a₁ a₂ b, r a₁ a₂ → f a₁ b = f a₂ b)
+  [Π a, decidable_pred (f a)] :
+  decidable (quot.lift_on₂ q₁ q₂ f ha hb) :=
+quot.lift₂.decidable_pred _ _ _ _ _ _ _
+
+instance (f : α → Prop) (h : ∀ a b, @setoid.r α s₁ a b → f a = f b) [decidable_pred f] :
+  decidable_pred (quotient.lift f h) :=
+quot.lift.decidable_pred _ _ _
+
+instance (f : α → β → Prop)
+  (h : ∀ a₁ a₂ b₁ b₂, @setoid.r α s₁ a₁ b₁ → @setoid.r β s₂ a₂ b₂ → f a₁ a₂ = f b₁ b₂)
+  [hf : Π a, decidable_pred (f a)] (q₁ : quotient s₁) :
+  decidable_pred (quotient.lift₂ f h q₁) :=
+λ q₂, quotient.rec_on_subsingleton₂' q₁ q₂ hf
+
+instance (q : quotient s₁) (f : α → Prop) (h : ∀ a b, @setoid.r α s₁ a b → f a = f b)
+  [decidable_pred f] :
+  decidable (quotient.lift_on q f h) :=
+quotient.lift.decidable_pred _ _ _
+
+instance (q₁ : quotient s₁) (q₂ : quotient s₂) (f : α → β → Prop)
```

Comment:
Is `s₁` in `{}` or `[]` here? The primed versions should use the former, the unprimed the latter.

Reply:
Actually, all versions here can use `{}`, because of note [implicit instance arguments].

Reply:
I was never convinced by that note, I suspect it might refer to a behavior in older version of Lean. Either way; are you saying this is already using `{}` binders?

Reply:
It already does, yes. But also I'm noticing that the organization is a bit of a mess. I guess I don't understand this file...

Reply:
The new home for the `quot` lemmas looks good. I think these ones should maybe still be separate into separate typeclass and implicit sections.